xdsl/dialects/csl.py

```diff
@@ -372,14 +533,14 @@ class ReturnOp(IRDLOperation):
 
     assembly_format = "attr-dict ($ret_val^ `:` type($ret_val))?"
 
-    traits = frozenset([HasParent(FuncOp), IsTerminator()])
+    traits = frozenset([HasParent(FuncOp, TaskOp), IsTerminator()])
 
     def __init__(self, return_val: SSAValue | Operation | None = None):
         super().__init__(operands=[return_val])
 
     def verify_(self) -> None:
         func_op = self.parent_op()
-        assert isinstance(func_op, FuncOp)
+        assert isinstance(func_op, FuncOp) or isinstance(func_op, TaskOp)
```

Comment:
This is handled by the `HasParent` traits verifier

Reply:
This was just to appease the type checker. I can use a `cast` instead

Reply:
Oh! I missed that you had more in the verifier! Sorry